### PR TITLE
Misc fixes

### PR DESCRIPTION
**Pull request by tdfischer**
A number of various fixes that the upcoming Sockets layer depends on

### CODE REVIEW

File: test/sandbox.cpp [hunk 1]
```diff
@@ -30,125 +22,89 @@ bool operator== (const Sandbox::SyscallCall& first, const Sandbox::SyscallCall&
...
+ExceptionIPC::onReadReady()
+{
+  char buf[2048];
+  int lineNum;
+  char filename[1024];
+  char message[1024];
+  memset (filename, 0, sizeof (filename));
+  memset (message, 0, sizeof (message));
+
+  read (parent, buf, sizeof (buf));
```
Comment (wilsonianb) [inline test/sandbox.cpp]: error: ignoring return value of ‘ssize_t read(int, void*, size_t)’, declared with attribute warn_unused_result [-Werror=unused-result]

File: test/vfs.cpp [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+
+  static void _runFunc(void* d) {
+    runner r = *static_cast<runner*>(d);
+    try {
+      ((r.self)->*(r.func))(nullptr);
+    } catch (CppUnit::Exception e) {
+      char buf[2048];
+      memset (buf, 0, sizeof (buf));
+      sprintf (buf, "%s\t%s\t%d", e.what(), e.sourceLine().fileName().c_str(), e.sourceLine().lineNumber());
+      write (42, buf, sizeof (buf));
```
Comment (wilsonianb) [inline test/vfs.cpp]: ignoring return value

File: test/sandbox.cpp (hunk 1 above)
Comment (tdfischer) [inline test/sandbox.cpp]: Fixed

File: test/vfs.cpp (hunk 2 above)
Comment (tdfischer) [inline test/vfs.cpp]: Fixed

**Comment by wilsonianb** [comment]
LGTM :+1:

**Pull request merged**